internal/query/projection/project_metadata_test.go

```diff
@@ -0,0 +1,185 @@
+	type args struct {
+		event func(t *testing.T) eventstore.Event
+	}
+	tests := []struct {
+		name   string
+		args   args
+		reduce func(event eventstore.Event) (*handler.Statement, error)
+		want   wantReduce
+	}{
+		{
+			name: "reduceMetadataSet",
+			args: args{
+				event: getEvent(
+					testEvent(
+						project.MetadataSetType,
+						project.AggregateType,
+						[]byte(`{
+						"key": "key",
+						"value": "dmFsdWU="
+					}`),
+					), project.MetadataSetEventMapper),
+			},
+			reduce: (&projectMetadataProjection{}).reduceMetadataSet,
+			want: wantReduce{
+				aggregateType: project.AggregateType,
+				sequence:      15,
+				executer: &testExecuter{
+					executions: []execution{
+						{
+							expectedStmt: "INSERT INTO projections.project_metadata (instance_id, project_id, key, resource_owner, creation_date, change_date, sequence, value) VALUES ($1, $2, $3, $4, $5, $6, $7, $8) ON CONFLICT (instance_id, project_id, key) DO UPDATE SET (resource_owner, creation_date, change_date, sequence, value) = (EXCLUDED.resource_owner, projections.project_metadata.creation_date, EXCLUDED.change_date, EXCLUDED.sequence, EXCLUDED.value)",
+							expectedArgs: []any{
+								"instance-id",
+								"agg-id",
+								"key",
+								"ro-id",
+								anyArg{},
+								anyArg{},
+								uint64(15),
+								[]byte("value"),
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "reduceMetadataRemoved",
+			args: args{
+				event: getEvent(
+					testEvent(
+						project.MetadataRemovedType,
+						project.AggregateType,
+						[]byte(`{
+						"key": "key"
+					}`),
+					), project.MetadataRemovedEventMapper),
+			},
+			reduce: (&projectMetadataProjection{}).reduceMetadataRemoved,
+			want: wantReduce{
+				aggregateType: project.AggregateType,
+				sequence:      15,
+				executer: &testExecuter{
+					executions: []execution{
+						{
+							expectedStmt: "DELETE FROM projections.project_metadata WHERE (instance_id = $1) AND (project_id = $2) AND (key = $3) AND (resource_owner = $4)",
+							expectedArgs: []any{
+								"instance-id",
+								"agg-id",
+								"key",
+								"ro-id",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "reduceMetadataRemovedAll",
+			args: args{
+				event: getEvent(
+					testEvent(
+						project.MetadataRemovedAllType,
+						project.AggregateType,
+						nil,
+					), project.MetadataRemovedAllEventMapper),
+			},
+			reduce: (&projectMetadataProjection{}).reduceMetadataRemovedAll,
+			want: wantReduce{
+				aggregateType: project.AggregateType,
+				sequence:      15,
+				executer: &testExecuter{
+					executions: []execution{
+						{
+							expectedStmt: "DELETE FROM projections.project_metadata WHERE (instance_id = $1) AND (project_id = $2)",
+							expectedArgs: []any{
+								"instance-id",
+								"agg-id",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "reduceOwnerRemoved(project removed)",
```

Comment:
This test is incorrect, owner removed is normally the organization.
So we would need an additional test for the project removed event.